corehq/apps/data_analytics/malt_generator.py

```diff
@@ -0,0 +1,75 @@
+from django.db import transaction
+
+from corehq.apps.app_manager.models import Application
+from corehq.apps.data_analytics.models import MALTRow
+from corehq.apps.domain.models import Domain
+from corehq.apps.smsforms.app import COMMCONNECT_DEVICE_ID
+from corehq.apps.sofabed.models import FormData
+
+
+class MALTTableGenerator(object):
+    """
+        Populates SQL table with data for given datespan
+        See MALTRow
+    """
+    _wam_pam_cache = {}
+
+    def __init__(self, datespan_object):
+        self.datespan = datespan_object
+
+    @transaction.atomic
```

Comment:
I don't think we want this to be atomic. For example, if something fails mid-way we would want to pick up and start from where we left off. Also none of the data affects any other data so it doesn't seem like atomicity is a priority.

Also, I have no idea what the performance implications would be of such a large and long-running transaction, but I'm guessing it could be problematic.